DEPRECATION.md

```diff
@@ -0,0 +1,81 @@
+# gRPC Swift 0.x deprecation
+
+## What is this document?
+
+This document outlines the current plan to deprecate the `0.x` release series of gRPC Swift.
+
+## What is happening?
+
+The `0.x` release series of the gRPC Swift project will be deprecated in favour of a new implementation.
```

Comment:
(Maybe) See my comment in README.md to provide more details in this first paragraph.